cleanlab/multiannotator.py

```diff
@@ -0,0 +1,436 @@
+        For details, predicted probabilities in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    method : str (Options: ["majority"])
+        The method used to aggregate labels from multiple annotators into a single consensus label. Default is by using the majority vote.
+        Options include:
+        - ``"majority"``: means consensus labels are reached via a simple majority vote among annotators, with ties broken via ``pred_probs``.
+
+    Returns
+    -------
+    consensus_labels: np.array
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+    """
+
+    valid_methods = ["majority"]
+    consensus_labels = np.empty(len(labels_multiannotator), dtype="int32")
+
+    # TODO: is there a more efficient/elegant way to do this?
+    if method == "majority":
+        mode_labels_multiannotator = labels_multiannotator.mode(axis=1)
+        for i in range(len(mode_labels_multiannotator)):
+            consensus_labels[i] = mode_labels_multiannotator.iloc[i][
+                pred_probs[
+                    i, mode_labels_multiannotator.iloc[i].dropna().astype(int).to_numpy()
+                ].argmax()
+            ]
+    else:
+        raise ValueError(
+            f"""
+            {method} is not a valid consensus method!
+            Please choose a valid consensus_method: {valid_methods}
+            """
+        )
+
+    return consensus_labels
+
+
+def get_annotator_agreement(
+    labels_multiannotator: pd.DataFrame,
+    consensus_labels: np.array,
+) -> np.array:
+    """Returns the fractions of annotators that agree with the consensus labels. Note that the
+    fraction for each example only considers the annotators that labeled that particular example.
+
+    Parameters
+    ----------
+    labels_multiannotator : pd.DataFrame
+        2D pandas DataFrame of multiple given labels for each example with shape (N, M),
+        where N is the number of examples and M is the number of annotators.
+        For more details, labels in the same format expected by the :py:func:`get_label_quality_scores_multiannotator <cleanlab.multiannotator.get_label_quality_scores_multiannotator>`.
+
+    consensus_labels : np.array
+        An array of shape ``(N,)`` with the consensus labels aggregated from all annotators.
+
+    Returns
+    -------
+    annotator_agreement : np.array
+        An array of shape ``(N,)`` with the fraction of annotators that agree with each consensus label.
+    """
+    return (
+        labels_multiannotator.assign(consensus_label=consensus_labels)
+        .apply(
+            lambda s: np.mean(s.drop("consensus_label") == s["consensus_label"]),
+            axis=1,
+        )
+        .to_numpy()
+    )
+
+
+# TODO: add other ways to the get quality score (methods that can be passed to get_label_quality_score)
+def get_quality_of_consensus(
```

Comment:
Let's make this helper method (with underscore prefix): `_get_quality_of_consensus()`

So we don't have to commit to supporting this exact API forever.